### PR TITLE
Expire trial tenants that have not logged in for 14 days

### PR DESCRIPTION
**Pull request by oyving**
* Add maintainer that expires trial tenants that have been idle for 14 days
* Created feature flag to exempt some tenants from expiry
* Extended SytemName parameter in ZoneApiMack to facilitate tests

FYI - @kkraune 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/CloudTrialExpirer.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.yahoo.vespa.hosted.controller.Controller;
+import com.yahoo.vespa.hosted.controller.api.integration.billing.PlanId;
+import com.yahoo.vespa.hosted.controller.tenant.LastLoginInfo;
+import com.yahoo.vespa.hosted.controller.tenant.Tenant;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CloudTrialExpirer extends ControllerMaintainer {
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/CloudTrialExpirer.java:16]: Needs javadoc and author.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/CloudTrialExpirerTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import com.yahoo.vespa.hosted.controller.tenant.LastLoginInfo;
+import com.yahoo.vespa.hosted.controller.tenant.Tenant;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class CloudTrialExpirerTest {
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/maintenance/CloudTrialExpirerTest.java:21]: Needs author.

**Review by mpolden — COMMENTED** [review]
This is fine for now, but I do wish we had a more general mechanism that covered all systems.

**Review by mpolden — APPROVED** [review]

**Comment by mpolden** [comment]
A test failed: ```[ERROR]   ControllerApiTest.testControllerApi:53 maintenance.json expected:<...eporter"},{"name":"C[]ontactInformationMai...> but was:<...eporter"},{"name":"C[loudTrialExpirer"},{"name":"C]ontactInformationMai...>```

`maintenance.json` must be updated.

**Comment by bratseth** [comment]
Do we need to do something in the console to help them understand what to do if they come back?

**Comment by oyving** [comment]
I think the the account page will show a message to contact support for the none subscription, but I'll make sure it's so.

**Comment by oyving** [comment]
Even better, I'll make sure it's a notification when they're in the none-subscription.

**Pull request merged**